Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -176,6 +176,9 @@ operator==(const TimeSeriesProperty<TYPE> &right) const {
   { // so vectors can go out of scope
     std::vector<DateAndTime> lhsTimes = this->timesAsVector();
     std::vector<DateAndTime> rhsTimes = right.timesAsVector();
+    if (lhsTimes.size() != rhsTimes.size()) {
```

Comment:
@peterfpeterson Could you tell me why an additional `if` is preferred to extending an existing one?